semeio/workflows/localisation/local_script_lib.py

```diff
@@ -853,3 +878,31 @@ def write_qc_parameter(
                 grid.write_grdecl(scaling_kw, file)
             # Increase parameter number to define unique parameter name
             cls.scaling_param_number = cls.scaling_param_number + 1
+
+
+def get_obs_from_ert(ert, expand_gen_obs_max_size):
```

Comment:
Is it possible to avoid the special casing of `GEN_OBS`?

Reply:
If not we should add some testing of it

Reply:
The reason why I defined this 'expand_gen_obs_max_size' was to avoid having to expand an GEN_OBS node into the list of type  nodename: index for index from 0 to size where the size sometimes can be very large (seismic data with 100.000's of values. The use of individual GEN_OBS observations is primarily meant for GEN_OBS nodes of moderate size. Examples I was thinking about was e.g geophysical data like gravimetry measurements or other cases where the number of observations in a GEN_OBS node is < 100. In this case it can be meaningful to setup localisations between individual observations and model parameters.  For seismic data, the obvious way of handling that is to split the seismic data into a moderate number of groups (e.g one per fault block or segment) and define one GEN_OBS node for each of these and use these without referring to individual observations within the GEN_OBS node. Sometimes, like for isolated segments, it can be relevant to setup localisation where a GEN_OBS node for a segment is used together with the model parameter values related to the same segment. I thought at least that there might be unnecessary overhead to work with lists of 100.000's of observations in the localisation script and introduced this expand_gen_obs_max_size parameter to handle ERT models where there exists GEN_OBS nodes having large number of observations and GEN_OBS nodes having moderate number or small number of observations ( < expand_gen_obs_max_size).  Do you have any better idea here?